src/select.c

```diff
@@ -378,6 +375,103 @@ int W32_CALL select (int nfds, fd_set *read_fds, fd_set *write_fds,
 }
 #endif
 
+/*
+ * POSIX poll() function.
+ */
+int W32_CALL poll (struct pollfd *p, int num, int timeout_ms)
+{
+  DWORD   timeout_at;
+  int     i, fd, ret = 0;
+
+  SOCK_DEBUGF (("\npoll: n=%d", num));
+
+  if (num < 0 || num > MAX_SOCKETS)
+  {
+    SOCK_DEBUGF ((", EINVAL"));
+    SOCK_ERRNO (EINVAL);
+    return (-1);
+  }
+
+  VERIFY_RW (p, num * sizeof (struct pollfd));
+
+  if (timeout_ms > 0)
+  {
+    SOCK_DEBUGF ((", timeout %lu ms", (u_long)timeout_ms));
+    timeout_at = set_timeout (timeout_ms);
+  }
+  else if (timeout_ms < 0)
+  {
+    SOCK_DEBUGF ((", no timeout"));
+  }
+
+  _sock_sig_setup();
+
+  /*
+   * Loop until specified timeout expires or event(s) satisfied.
+   */
+  while (1)
+  {
+    /* Not safe to run sock_daemon() (or other "tasks") now
+     */
+    _sock_crit_start();
```

Comment:
Looking at this again, I don't think this is necessary at all.  If the socket gets deleted it just returns `POLLNVAL`.

In fact, `_sock_crit_stop()` will now run the daemon on every loop so this only hurts performance.